app/wasm/queries.go

```diff
@@ -26,6 +27,26 @@ func NewQueryPlugin(
 	}
 }
 
+func (qp QueryPlugin) GetFullDenom(ctx sdk.Context, contract string, subDenom string) (*string, error) {
+	// Address validation
+	contractAddress, err := sdk.AccAddressFromBech32(contract)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "address from bech32")
+	}
+	err = sdk.VerifyAddressFormat(contractAddress)
```

Comment:
Huh, this is new.
I have always only done `sdk.AccAddressFromBech32` to parse and assume it would error on any invalid address.

What checks does `sdk.VerifyAddressFormat` do and should we add it everywhere in x/wasm as well?

Reply:
It uses a validator that is being passed to it. This ends up being

https://github.com/CosmWasm/wasmd/blob/e7a3ac346df8f66fa085f5d5605f540a9deafc4d/x/wasm/types/types.go#L325-L332

I'm now thinking that this check is redundant, and converting to address from bech32 should be enough.